utils/submit_tracks.py

```diff
@@ -0,0 +1,197 @@
+#! /usr/bin/env python3
+
+import sys
+import json
+import requests
+
+track_api_root = 'https://staging-2020.ensembl.org/api/tracks'
+
+def submit_track(track_data, retry=False):
+  try:
+    request = requests.post(f"{track_api_root}/track", json=track_data)
+  except requests.exceptions.ConnectTimeout:
+    if(retry):
+      print(f"No luck, bailing out.")
+      exit(1)
+    else:
+      print(f"Connection timed out. Retrying...")
+      submit_track(track_data, True)
+  
+  msg = request.content.decode()
+  if request.status_code != 201:
+    if 'Track already exists' in msg:
+      print(f"Track {track_data['trigger'][-1]} already exists, skipping.")
+      return
+    print(f"Error submitting track ({request.status_code}): {msg}")
+    print(f"Track payload: {track_data}")
+    exit(1)
+  print(msg) #expected response: {"track_id": "some-uuid"}
+
+def parse_csv(path):
+  if not path or not path.endswith('.csv'):
+    print('Input CSV file missing')
+    exit(1)
+  with open(path) as f:
+    lines = f.readlines()
+  lines.pop(0) #remove header
+  return [line.strip().split(',') for line in lines]
+
+mode, input = (sys.argv[1], sys.argv[2]) if len(sys.argv) > 2 else (None, None)
+
+if(mode == 'variation'):
+  if not input or not input.endswith('.json'):
+    print('Input JSON file missing')
+    exit(1)
+  with open(input) as f:
+    input_data = json.load(f)
+
+  print(f"Submitting {len(input_data)} variation tracks:")  
+  for uuid in input_data:
+    print(f"Species {uuid}:")
+    for field in ['label','datafiles','description','source']:
+      if field not in input_data[uuid] or not input_data[uuid][field]:
+        print(f"Missing field in input JSON: {uuid}->{field}")
+        exit(1)
```

Comment:
Are you assuming if it is missing in 1 object it is missing on all the objects ?
exit ? or break ? I this will exit the script without even iterating over subsequent objects in json file.

Reply:
Yes, this is a simple test for the input file to break the script (user supplied worng file by accident).

Reply:
It might be the case that it's just missing for one genome but it might well be there for other genomes. The point is we should not hold back tracks for other genome just because one is not good. Think about trying to load tracks for 241 genomes and only few genomes ( say 6) input data is not appropriate. Would you hold loading track for 235 ?

Reply:
For 241, I would say yes the data is broken, we need to fix data and restart script. We should also flush the database (we don't win more than 3 minutes for loading all tracks vs only failed tracks), but the script also supports resuming by skipping genomes/tracks that have already been loaded and continuing from where it left off.

Post-241, we need to make the loading srcipt/procedure more complex to properly handle gaps in data: we need to flag genomes/tracks that failed, decide what to do with existing data (e.g. do we clean existing tracks for partially-loaded species), how to fill gaps after fixing the data (e.g. where to keep the uuids for skipped species), add data-check stage (check all track data before starting to load db), handle partial updates to existing tracks etc.

Reply:
> we don't win more than 3 minutes for loading all tracks vs only failed tracks

It's not just about 3 minutes it takes to load/reload. It's about the other things which will be blocked by not having tracks loaded for which we have acceptable data. 
Think about you won't be able to view any tracks on genome browser even though genome browser files are available, variation, regulation files are available, thoas has been loaded with all the data for that genome. 
If it is just the track API which is blocking critical stuff. I would block the tracks for the genome for which we have acceptable data.

> We should also flush the database

We should focus on incremental update rather than delete /reload everything.

> Post-241, we need to make the loading srcipt/procedure more complex to properly handle gaps in data

It can be done now. Just print the genome_id instead of exit(1). put break statement and continue and you will have the list of genomes for which tracks hasn't been loaded

Reply:
I added the skip/continue part.  I'll leave proper process resuming for the future (loading data only for the genomes skipped in the previous run). For now, the script will start data loading from the beginning, skipping tracks that Track API finds existing in the database.